portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -42,6 +46,7 @@
   private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
   private Gson gson = new Gson();
   private Query commentsQuery = new Query(COMMENT_ENTITY).addSort("timestamp", SortDirection.DESCENDING);
+  private UserService userService = UserServiceFactory.getUserService();
```

Comment:
nit: these fields can actually all be made `final`

(it's best to make fields `final` whenever possible as it allows readers to reason about when/if they can change, without having to read the whole class)